src/Lightning/Network/Protocol/Transport/HandshakeNoiseProtocol.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using Network.Protocol.Transport.Noise;
+using NoiseProtocol;
+using LightningNetworkConfig = Network.Protocol.Transport.Noise.LightningNetworkConfig;
+
+namespace Network.Protocol.Transport
+{
+   public class HandshakeWithNoiseProtocol : IHandshakeProtocol
+   {
+      private const int HEADER_LENGTH = 18;
+
+      public byte[]? RemotePubKey { get; set; }
+      public string LocalPubKey { get; set; }
```

Comment:
LocalPubKey  needs to also be bytes no?

Reply:
Removed local  public for now, we'll add if needed.